doc/source/ray-core/walkthrough.rst

```diff
@@ -2,125 +2,135 @@
 
 Starting Ray
 ------------
+First, let's include all modules needed for this tutorial and start a local Ray cluster with :ref:`ray.init() <ray-init-ref>`:
 
-You can start Ray on a single machine by adding this to your code.
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __starting_ray_start__
+    :end-before: __starting_ray_end__
 
 .. note::
 
   In recent versions of Ray (>=1.5), ``ray.init()`` will automatically be called on the first use of a Ray remote API.
 
-.. tabbed:: Python
-
-    .. code-block:: python
-
-        import ray
-
-        # Start Ray.
-        ray.init()
-
-        ...
-
-.. tabbed:: Java
-
-    .. code-block:: java
-
-        import io.ray.api.Ray;
-
-        public class MyRayApp {
-
-            public static void main(String[] args) {
-                // Start Ray runtime. If you're connecting to an existing cluster, you can set
-                // the `-Dray.address=<cluster-address>` java system property.
-                Ray.init();
-                ...
-            }
-        }
-
-.. tabbed:: C++
-
-    .. code-block:: c++
-
-        // Run `ray cpp --show-library-path` to find headers and libraries.
-        #include <ray/api.h>
-
-        int main(int argc, char **argv) {
-            // Start Ray runtime. If you're connecting to an existing cluster, you can set
-            // the `RAY_ADDRESS` env var.
-            ray::Init();
-            ...
-        }
-
-
-Ray will then be able to utilize all cores of your machine. Find out how to configure the number of cores Ray will use at :ref:`configuring-ray`.
-
-To start a multi-node Ray cluster, see the :ref:`cluster setup page <cluster-index>`.
-
-Using Tasks, Actors, and Objects
---------------------------------
 
-Click through below to walk through using Ray's key concepts: Tasks, Actors, and Objects.
+Defining the Progress Actor
+---------------------------
+Next, we will define a Ray actor that can be called by sampling tasks to update progress.
+Ray actors are essentially stateful services that anyone with an instance (a handle) of the actor can call its methods.
+
+.. literalinclude:: doc_code/getting_started.py
+    :language: python
+    :start-after: __defining_actor_start__
+    :end-before: __defining_actor_end__
+
+We define a Ray actor by decorating a normal Python class with :ref:`ray.remote() <ray-remote-ref>`.
+The progress actor has ``report_progress()`` method that will be called by sampling tasks to update their progress individually
+and ``get_progress()`` method to get the overall progress.
+
+Defining the Sampling Task
+--------------------------
+Next, we will define a Ray task that does the sampling up to the given times and returns the number of samples that are inside the circle.
+Ray tasks are stateless functions that will be executed asynchronously and in parallel.
```

Comment:
Use present tense since this is always true.

Ray tasks are stateless functions. They are scheduled on Ray cluster workers' nodes, execute asynchronously, and run in parallel.